### PR TITLE
[fix]: [CDS-53600]: Fix error message on webhook registration failure

### PR DESCRIPTION
**Pull request by rathodmeetsatish**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/44852)

### CODE REVIEW

**Review by vcalasansh — DISMISSED** [review]
> Reviewed 6 of 11 files at r1, 13 of 13 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44852)* status: 19 file(s) to review. 1 of 2 approvals obtained.

File: 120-ng-manager/src/main/java/io/harness/ng/webhook/errorhandler/ScmApiErrorHandlingHelper.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import lombok.experimental.UtilityClass;
+import lombok.extern.slf4j.Slf4j;
+
+@UtilityClass
+@Slf4j
+@OwnedBy(SPG)
+public class ScmApiErrorHandlingHelper {
+  private static final Map<ConnectorType, Set<Integer>> failureErrorCodeExceptions =
+      ImmutableMap.<ConnectorType, Set<Integer>>builder()
+          .put(ConnectorType.AZURE_REPO, new HashSet<>(Collections.singletonList(203)))
```
Comment (sribalij) [inline 120-ng-manager/src/main/java/io/harness/ng/webhook/errorhandler/ScmApiErrorHandlingHelper.java:37]: Why 203 specifically?

File: 120-ng-manager/src/main/java/io/harness/ng/webhook/services/impl/WebhookServiceImpl.java [hunk 2]
```diff
@@ -73,19 +98,38 @@ public UpsertWebhookResponseDTO upsertWebhook(UpsertWebhookRequestDTO upsertWebh
...
+          createWebhookResponse.getStatus(), scmConnector, createWebhookResponse.getError(), upsertWebhookRequestDTO);
       UpsertWebhookResponseDTO response = UpsertWebhookResponseDTO.builder()
                                               .webhookResponse(createWebhookResponse.getWebhook())
                                               .error(createWebhookResponse.getError())
                                               .status(createWebhookResponse.getStatus())
                                               .build();
       log.info("Upsert Webhook Response : {}", response);
-      return response;
+      upsertWebhookResponseDTO = response;
+    } catch (ExplanationException ex) {
```
Comment (sribalij) [inline 120-ng-manager/src/main/java/io/harness/ng/webhook/services/impl/WebhookServiceImpl.java:110]: Making sure the order of catching is right?

**Review by sribalij — DISMISSED** [review]

File: 120-ng-manager/src/main/java/io/harness/ng/webhook/errorhandler/ScmApiErrorHandlingHelper.java (hunk 1 above)
Comment (rathodmeetsatish) [reply 120-ng-manager/src/main/java/io/harness/ng/webhook/errorhandler/ScmApiErrorHandlingHelper.java]: git sync team would have observed the code as error code in some case with azure repo. Removed it as it is not relevent to upsert webhhok operation

File: 120-ng-manager/src/main/java/io/harness/ng/webhook/services/impl/WebhookServiceImpl.java (hunk 2 above)
Comment (rathodmeetsatish) [reply 120-ng-manager/src/main/java/io/harness/ng/webhook/services/impl/WebhookServiceImpl.java:110]: yes

**Review by mohitg0795 — APPROVED** [review]
> Reviewed 5 of 11 files at r1, 9 of 13 files at r3, 4 of 5 files at r4, 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44852)* status: 6 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained.

**Comment by rathodmeetsatish** [comment]
trigger messagecheck

**Review by vcalasansh — APPROVED** [review]
> Reviewed 4 of 5 files at r4, 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44852)* status: 2. unresolved discussions.

**Review by vcalasansh — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44852)* status: 2. unresolved discussions.

**Review by sribalij — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44852)* status: :shipit: complete! ✓

**Pull request merged**